### PR TITLE
Add Kpt + Gcloud Image Definition

### PR DESCRIPTION
**Pull request by runewake2**
See: #1506

This introduces a new container image that includes `gcloud` alongside other programs configured with `kpt`. The image will be released as `kpt-gcloud` into the same container registry as the kpt container (`kpt-dev`). The image is based off of gcr.io/google.com/cloudsdktool/cloud-sdk:alpine

This image is significantly larger than the `kpt` image already published at gcr.io/kpt-dev/kpt but smaller than the larger general purpose `cloud-sdk` images that are available with `kpt`.

***

This PR also updates the kpt releasing README with steps that will allow you to run and validate Cloud Build releases of `kpt` against your local fork.

Hardcoded github owner values have been removed from the kpt Goreleaser config to prevent publishing new releases to the `GoogleContainerTools/kpt` repo unintentionally. (note: this is still possible with wrong parameters and notes have been included to prevent this as best as possible)

> Note: A followup change to include the image on kpt.dev will follow the first release of these new images.

### CODE REVIEW

File: gcloud/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Google LLC
```
Comment (mortent) [inline gcloud/Dockerfile:1]: Would it be better to put this inside the `release` directory? It seems strange to create a new top-level `gcloud` directory.

File: release/tag/goreleaser.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ builds:
     goarch:
       - amd64
       - arm64
-    ldflags: -s -w -X github.com/GoogleContainerTools/kpt/run.version={{.Version}}
+    ldflags: -s -w -X github.com/{{ .Env._GITHUB_USER }}/kpt/run.version={{.Version}}
```
Comment (mortent) [inline release/tag/goreleaser.yaml:26]: I don't think we should change this. The package path would be the same regardless of whether it run on the `GoogleContainerTools` repo or somewhere else right?

**Comment by frankfarzan** [comment]
In the PR description, can you specify the size of the two container images?

File: release/gcloud/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Google LLC
```
Comment (frankfarzan) [inline release/gcloud/Dockerfile:1]: For consistency, the existing top-level Dockerfile should be moved to release directory as well. This can be a separate PR

File: release/gcloud/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ARG KPT_VERSION=unknown
+
+WORKDIR /go/src/
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+COPY . .
+RUN go build -v -o /usr/local/bin/kpt -ldflags="-s -w -X github.com/GoogleContainerTools/kpt/run.version=$KPT_VERSION" ./
+
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
```
Comment (frankfarzan) [inline release/gcloud/Dockerfile:28]: Should probably use a pinned version instead. We don't want to have this change without our explicit consent.

**Review by frankfarzan — CHANGES_REQUESTED** [review]

File: release/gcloud/Dockerfile (hunk 3 above)
Comment (runewake2) [reply release/gcloud/Dockerfile:1]: Ack. Will open that PR after this is merged to avoid conflicts with the merge/squash of this change.

**Review by frankfarzan — APPROVED** [review]

File: release/gcloud/Dockerfile (hunk 4 above)
Comment (runewake2) [reply release/gcloud/Dockerfile]: I've pinned this to `332.0.0-alpine` and updated our releasing steps to include updates to the base image versions prior to release.

**Review by mortent — APPROVED** [review]
